### PR TITLE
Preview Providers now compile correctly

### LINKED ISSUE #156: Allow `thenProceed` in preview providers
### Is your feature request related to a problem? Please describe.
I want to use previews to view my `FlowRepresentable` as it'll appear in a workflow, even if that's just a single item workflow. However, right now `thenProceed` is not available.

### Describe the solution you'd like
Either make `thenProceed` globally available so we quit having one-off "I need it added" or add it to `PreviewProvider`.

### Describe alternatives you've considered
I could write my own extension for this, but it should just work out of the box. Additionally, I suppose we could wrap `thenProceed` in some new protocol? However, I don't like the idea that I have to conform to a protocol just to define a workflow. There may be something to investigating resultBuilders in a future version of Swift where they allow those scoped functions. I've seen proposals about it, but for now I don't think we should go that route.

### PR DESCRIPTION
**Pull request by Tyler-Keith-Thompson**
## Checklist:
- [x] Did you sign the [Contributor License Agreement](https://cla-assistant.io/wwt/SwiftCurrent)?
- [x] Is the linter reporting 0 errors?
- [x] Did you comply with our [styleguide](https://github.com/wwt/SwiftCurrent/blob/main/.github/STYLEGUIDE.md)?
- [x] Is there [adequate test coverage](https://github.com/wwt/SwiftCurrent/blob/main/.github/CONTRIBUTING.md#test-etiquette) for your new code?
- [x] Does the CI pipeline pass?
- [x] Did you [update the documentation](https://github.com/wwt/SwiftCurrent/blob/main/.github/CONTRIBUTING.md#documentation)?
- [x] Did you [change the public API](https://github.com/wwt/SwiftCurrent/blob/main/.github/CONTRIBUTING.md#public-api)?
- [x] Have you done everything you can to make sure that errors that can occur are compile-time errors, and if they have to be runtime do you have adequate tests and documentation around those runtime errors? [For more details](https://github.com/wwt/SwiftCurrent/blob/main/.github/CONTRIBUTING.md#errors).

----

### If Public API Has Changed:
- [ ] Did you deprecate (rather than remove) any old methods/variables/etc? [Our philosophy for deprecation](https://github.com/wwt/SwiftCurrent/blob/main/.github/CONTRIBUTING.md#deprecation).
- [x] Have you done the best that you can to make sure that the compiler guides people to changing to the new API? (Example: the renamed attribute)
- [x] If necessary, have you tested the upgrade path for at least N-1 versions? For example, if data persists between v1 and v2 then that upgrade should be tested and as easy as we can make it.

### CODE REVIEW

**Comment by morganzellers** [comment]
> * There is a breaking change in here. You used to be able to call `self.thenProceed`, which referred to a local method. Now that the method is global and the local methods are removed this no longer works and results in an error. @morganzellers what do you want to do about that?

I think we should version bump then. Does this force us to do a major version bump, or could we get away with bumping to v4.6.0?

File: Sources/SwiftCurrent_SwiftUI/Extensions/ThenProceedExtensions.swift [hunk 1]
```diff
@@ -20,143 +19,56 @@ private func verifyWorkflowIsWellFormed<LHS: FlowRepresentable, RHS: FlowReprese
...
-        return WorkflowItem(VC.self) { nextItem() }
-    }
-    #endif
+/**
+ Adds an item to the workflow; enforces the `FlowRepresentable.WorkflowOutput` of the previous item matches the args that will be passed forward.
+ - Parameter with: a `FlowRepresentable` type that should be presented.
+ - Parameter nextItem: a closure returning the next item in the `Workflow`.
+ - Returns: a new `WorkflowItem` with the additional `FlowRepresentable` item.
+ - NOTE: Should be called inside a `WorkflowLauncher` initializer.
+ - IMPORTANT: Not for use in UIKit, unless you're doing SwiftUI interop.
```
Comment (morganzellers) [inline Sources/SwiftCurrent_SwiftUI/Extensions/ThenProceedExtensions.swift:67]: Nice nice

**Review by morganzellers — APPROVED** [review]

**Comment by morganzellers** [comment]
I feel like I'm okay just gating this for our next version bump. It feels far away, but this isn't worth bumping to 5.0 now

**Comment by Tyler-Keith-Thompson** [comment]
Agreed. Should we create a 5.0 branch and merge it there?

**Comment by morganzellers** [comment]
> Agreed. Should we create a 5.0 branch and merge it there?

I like that idea, I can do that

**Pull request merged**